### PR TITLE
Bugfix/search items

### PR DESCRIPTION
**Pull request by iqmulator**

### CODE REVIEW

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundResult.razor [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@namespace ActualChat.UI.Blazor.App.Components
+@using ActualChat.Search
+@{
+    if (ReferenceEquals(FoundItem, null))
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundResult.razor]: When FoundItem can be null?

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundChatListItem.razor [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                InitialValue = Model.Loading,
+                Category = ComputedStateComponent.GetStateCategory(t),
+            });
+
+    protected override async Task<Model> ComputeState(CancellationToken cancellationToken)
+    {
+        // We intentionally use .ConfigureAwait(false) here: this is one of the most frequently called methods.
+        // Properties are copied to local vars because of this.
+        Place? place = null;
+        var foundItem = FoundItem;
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundChatListItem.razor:70]: According to FoundResult component FoundItem parameter value is never null.

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundContactListItem.razor [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+@code {
+    [Inject] private ChatUIHub Hub { get; init; } = null!;
+    private ChatUI ChatUI => Hub.ChatUI;
+    private IAuthors Authors => Hub.Authors;
+    private AccountUI AccountUI => Hub.AccountUI;
+    private Session Session => Hub.Session();
+
+    [Parameter] public FoundItem? FoundItem { get; set; }
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundContactListItem.razor]: According to FoundResult component FoundItem parameter value is never null.

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundMessageListItem.razor [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+@code {
+    [Inject] private ChatUIHub Hub { get; init; } = null!;
+    private ChatUI ChatUI => Hub.ChatUI;
+    private IAuthors Authors => Hub.Authors;
+    private AccountUI AccountUI => Hub.AccountUI;
+    private Session Session => Hub.Session();
+    private IPlaces Places => Hub.Places;
+
+    [Parameter] public FoundItem? FoundItem { get; set; }
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundMessageListItem.razor]: According to FoundResult component FoundItem parameter value is never null.

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+@code {
+    [Inject] private ChatUIHub Hub { get; init; } = null!;
+    private ChatUI ChatUI => Hub.ChatUI;
+    private IAuthors Authors => Hub.Authors;
+    private IPlaces Places => Hub.Places;
+    private Session Session => Hub.Session();
+
+    [Parameter] public FoundItem? FoundItem { get; set; }
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor]: According to FoundResult component FoundItem parameter value is never null.

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundResult.razor (hunk 1 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundResult.razor]: Does FoundChatList supply Null values?

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundResult.razor (hunk 1 above)
Comment (frolyo) [reply src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundResult.razor]: it's neve null

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundContactListItem.razor [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+@using ActualChat.Contacts
+@inherits ComputedRenderStateComponent<FoundContactListItem.Model>
+@{
+    var m = State.Value;
+    if (m == Model.None)
+        return;
+
+    var chatState = m.ChatState;
+    var chat = chatState.Chat;
+    if (ReferenceEquals(chat, null))
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundContactListItem.razor]: Chat is never null here

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundMessageListItem.razor [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+@using ActualChat.Contacts
+@inherits ComputedRenderStateComponent<FoundMessageListItem.Model>
+@{
+    var m = State.Value;
+    if (m == Model.None)
+        return;
+
+    var chatState = m.ChatState;
+    var chat = chatState.Chat;
+    if (ReferenceEquals(chat, null) || ReferenceEquals(m.Entry, null))
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundMessageListItem.razor]: Chat is never null here

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundMessageListItem.razor [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+        var chatState = await ChatUI.GetState(chatId, false, cancellationToken).ConfigureAwait(false);
+        if (chatState == null)
+            return Model.None;
+
+        var placeId = chatId.PlaceChatId.PlaceId;
+        if (!placeId.IsNone)
+            place = await Places.Get(Session, placeId, cancellationToken).ConfigureAwait(false);
+
+        var entry = entryId.IsNone ? chatState.Info.LastTextEntry : await ChatUI.GetEntry(entryId, cancellationToken);
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundMessageListItem.razor]: Can entryId be none for found message?

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        return;
+}
+
+@if (m == Model.Loading) {
+    <ChatListItemSkeleton />
+} else {
+    <div class="found-result place">
+        <div class="c-content" data-href="@m.Link">
+            <div class="c-container">
+                <ChatIcon Chat="chatState.Chat"/>
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor:21]: We should show place icon here, not chat icon. See NavbarPlaceButtons.
You can change now or later.
After this change ChatState will not be needed.

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+        var allAuthorIds = await Places.ListAuthorIds(Session, place.Id, cancellationToken).ConfigureAwait(false);
+        var welcomeChatId = await Places.GetWelcomeChatId(Session, chatId.PlaceChatId.PlaceId, cancellationToken).ConfigureAwait(false);
+
+        return new() {
+            ChatState = chatState,
+            TitleMatch = titleMatch.IsEmpty ? SearchMatch.New(chatState.Chat.Title) : titleMatch,
+            Place = place,
+            MemberCount = allAuthorIds.Count,
+            Link = Links.Chat(welcomeChatId),
```
Comment (crui3er) [inline src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor]: Use Links.PlaceInfo.
Welcome chat is not needed.

**Review by crui3er — CHANGES_REQUESTED** [review]

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor (hunk 10 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor]: I've changed it myself.

File: src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor (hunk 9 above)
Comment (crui3er) [reply src/dotnet/UI.Blazor.App/Components/ChatList/FoundResult/FoundPlaceListItem.razor:21]: I've changed it myself.

**Pull request merged**